torch_audiomentations/augmentations/impulse_response.py

```diff
@@ -0,0 +1,37 @@
+import numpy as np
+import random
+import torch
+from torch_audiomentations.core.transforms_interface import BasicTransform
+from torch_audiomentations.utils.convolution import convolve
+from torch_audiomentations.utils.file import find_files, load_audio
+
+
+class ImpulseResponse(BasicTransform):
+    """
+    Convolves an input signal with an impulse response.
+    """
+
+    def __init__(self, ir_path, convolve_mode="full", p=0.5):
+        super(ImpulseResponse, self).__init__(p)
+        self.ir_path = find_files(ir_path)
+        self.convolve_mode = convolve_mode
+
+    def randomize_parameters(self, samples, sample_rate):
+        super(ImpulseResponse, self).randomize_parameters(samples, sample_rate)
+
+        if self.parameters["should_apply"]:
+            self.parameters["ir_file_path"] = random.choice(self.ir_path)
+
+    def apply(self, samples, sample_rate):
+        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
```

Comment:
I think `samples` should already be a torch tensor when passed to apply, so we can chain multiple transforms without having to convert between torch and numpy. I also think that it should return a torch tensor, not a numpy ndarray

Reply:
@iver56 - Just addressed the comments for: 1) `samples` being a `torch` tensor, and 2) switching from `ImpulseResponse` to `ApplyImpulseResponse`.

Tests and other files are updated to support these changes.